pkg/virtctl/imageupload/imageupload.go

```diff
@@ -546,6 +562,14 @@ func createUploadDataVolume(client kubecli.KubevirtClient, namespace, name, size
 		return nil, err
 	}
 
+	// We check if the storageClass exists before attempting to create the dataVolume
```

Comment:
We explicitly didn't check for storage class as it is entirely valid to have a kubernetes cluster without a storage class, but with manually provisioned PVs. minikube or CRC are good examples of this. This change would make upload no longer work on those.

Reply:
good catch, I forgot about this when reviewing!

Reply:
I've been thinking about handling this in a different way (just print some message when no storage class is defined, maybe) but maybe we should keep this behavior. In that code, I just return error when a storage class is defined as a flag but it doesn't exist. 

The case @awels is describing should still work since no storage class would be specified in the flags, so manually provisioned PVs would work. What do you think?

Reply:
If a storage class is specified in the flags but doesn't exist, the command just hangs without providing useful information.

Reply:
If the user specifies a storage class in a flag AND that storage class doesn't exist then it would make sense to generate this error. But if the user doesn't specify a storage class, we shouldn't assume there is one and generate an error.

Reply:
That's what we do there since we are only checking `storageClass` (the global, user-defined flag).

```go

// We check if the storageClass exists before attempting to create the PVC
if storageClass != "" { 
    _, err := client.StorageV1().StorageClasses().Get(context.Background(), storageClass, metav1.GetOptions{})
    if err != nil {
	    return nil, err
    }
}
```

I will improve the comment to specify that we are only checking the user-defined storage class.